### PR TITLE
Added the capability to register custom loggers (for Issue #626)

### PR DESCRIPTION
**Pull request by scott1y**
Hi Nick (and/or other guys),
I have made changes for Issue #626.  Please review.  Thanks!

### CODE REVIEW

File: red/log.js [hunk 1]
```diff
@@ -73,11 +83,19 @@ var log = module.exports = {
 
     init: function(settings) {
         logHandlers = [];
-        var consoleSettings = {};
-        if (settings.logging) {
-            consoleSettings = settings.logging.console || {};
+        var loggerSettings = {};
+        var keys = Object.keys(settings.logging);
```
Comment (knolleary) [inline red/log.js]: This assumes `settings.logging` exists - if it doesn't exist, `Object.keys` throws an exception (which is what has broken the tests).

**Comment by knolleary** [comment]
Thanks @scott1y 

On the whole, looks good. Have added an inline-comment on a bit that is breaking the unit tests.

I'm not sure it should be using the console output for loggers that don't provide their own handler. I think the approach should be (but happy to discuss):

1. If `settings.logging` is null, default to a console logger
2. otherwise, for each named logger under `settings.logging`:
    1. if the name is `console` - add an instance of the console logger
    2. otherwise, if no handler is specified, ignore the entry
    3. otherwise create the log handler wrapper for it

**Comment by scott1y** [comment]
Hi Nick,  I have corrected based on above your comments.  (Also, sorry that I forgot to delete some commented lines.)  Please take a look.  Thank you very much for your valuable comments!

**Comment by knolleary** [comment]
Thanks, looks good at a glance - will pull it down and try it out.

**Pull request merged**